timelock-api/src/main/conjure/timelock-api.yml

```diff
@@ -217,6 +217,40 @@ types:
       LeaderTimes:
         fields:
           leaderTimes: map<Namespace, LeaderTime>
+      TimestampName:
+        alias: string
+        safety: safe
+      NamedMinTimestampLeaseRequest:
+         fields:
+           timestampName: TimestampName
+           requestId:
+             type: uuid
+             safety: safe
+           numFreshTimestamps:
+             type: integer
+             safety: safe
+      NamedMinTimestampLeaseRequests:
+         alias: list<NamedMinTimestampLeaseRequest>
+      MultiClientNamedMinTimestampLeaseRequest:
+         alias: map<Namespace, NamedMinTimestampLeaseRequests>
+      NamedMinTimestampLeaseResponse:
+         fields:
+           minLeased: Long
+           lock: LockToken
+           lease: Lease
+           freshTimestamps: ConjureTimestampRange
+      NamedMinTimestampLeaseResponses:
+         alias: list<NamedMinTimestampLeaseResponse>
+      MultiClientNamedMinTimestampLeaseResponse:
+         alias: map<Namespace, NamedMinTimestampLeaseResponses>
+      GetMinLeasedNamedTimestampRequests:
+         alias: set<TimestampName>
+      MultiClientGetMinLeasedNamedTimestampRequest:
```

Comment:
Will you want to define some max limits here?

Reply:
This should not get too big for our practical purposes right?

The number of possible clients should not be more than 200, and the number of named timestamps won't also be too big